pkg/collector/nakadi_collector.go

```diff
@@ -72,14 +71,40 @@ func NewNakadiCollector(_ context.Context, nakadi nakadi.Nakadi, hpa *autoscalin
 		return nil, fmt.Errorf("metric-type must be either '%s' or '%s', was '%s'", nakadiMetricTypeConsumerLagSeconds, nakadiMetricTypeUnconsumedEvents, metricType)
 	}
 
+	// Either subscription-id or filtering via owning-application,
+	// event-types, and consumer-group is supported. If all are defined
+	// then only subscription-id is used and the rest of the fields are
+	// ignored.
+	subscriptionFilter := &nakadi.SubscriptionFilter{}
+	if subscriptionID, ok := config.Config[nakadiSubscriptionIDKey]; ok {
+		subscriptionFilter.SubscriptionID = subscriptionID
+	}
+
+	if owningApplication, ok := config.Config[nakadiOwningApplicationKey]; ok {
+		subscriptionFilter.OwningApplication = owningApplication
+	}
+
+	if nakadiEventTypes, ok := config.Config[nakadiEventTypesKey]; ok {
+		eventTypes := strings.Split(nakadiEventTypes, ",")
+		subscriptionFilter.EventTypes = eventTypes
+	}
+
+	if consumerGroup, ok := config.Config[nakadiConsumerGroupKey]; ok {
+		subscriptionFilter.ConsumerGroup = consumerGroup
+	}
+
+	if subscriptionFilter.SubscriptionID == "" && (subscriptionFilter.OwningApplication == "" && len(subscriptionFilter.EventTypes) == 0 && subscriptionFilter.ConsumerGroup == "") {
+		return nil, fmt.Errorf("either subscription-id or one of [owning-application, event-types, consumer-group] must be specified on the metric")
+	}
```

Comment:
We actually need the whole triple of [owning-application, event-types, consumer-group] to determine the subscription ID uniquely. (While consumer-group is optional, it is nowadays considered good form to always have it, and I guess we can accept if this doesn't support consumer-group-less subscriptions.)

```suggestion
	if subscriptionFilter.SubscriptionID == "" && (subscriptionFilter.OwningApplication == "" || len(subscriptionFilter.EventTypes) == 0 || subscriptionFilter.ConsumerGroup == "") {
		return nil, fmt.Errorf("either subscription-id or all of [owning-application, event-types, consumer-group] must be specified on the metric")
	}
```